.github/workflows/ci.yml

```diff
@@ -1,12 +1,48 @@
-name: Butte
-on: [push]
+on: push
+
+name: Continuous Integration
 
 jobs:
-  build:
-    name: Cargo Test
+  ci:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - beta
+          - nightly
+          - 1.31.0  # MSRV
+
     steps:
-      - name: 'Checkout the code'
+      - name: Checkout
         uses: actions/checkout@v1
-      - name: 'Cargo test'
-        run: cargo test
+
+      - name: Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+          components: rustfmt, clippy
+
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+
+      - name: Test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+      - name: Format
```

Comment:
Should we run this check first? I assume that it'd fail faster than waiting for build/test first only to fail a format check.

Reply:
Good idea, done